test/flatten.ts

```diff
@@ -139,4 +141,18 @@ describe('flatten', () => {
       )
     );
   });
+
+  it('should unsubscribe inner stream when sink unsubscribes', () => {
+    let taken = 0;
+
+    pipe(
+      of(1),
+      map(() => fromArray([1, 2, 3, 4])),
+      flatten,
+      take(2),
+      subscribe(() => taken++)
+    );
+
+    assert.strictEqual(taken, 2);
+  });
```

Comment:
Does this test case actually test that it 'unsubscribes inner stream when sink unsubscribes'? It doesn't seem clear to me that this snippet tests that.

Reply:
Well, I've tried to construct a test case resembling real usage - because of that it might be harder to see if it tests the right thing, but imho it's a better test in the end. If you think otherwise I can change it. 

If we remove my fix then this test fails, because the inner `fromArray` "pumps" its all items (it cannot be stopped/unsubscribed by `take(2)`).

Reply:
I see! Thanks